File: Leetcode/cpp/easy/485.cpp
/*
 * @lc app=leetcode id=485 lang=cpp
 *
 * [485] Max Consecutive Ones
 */


// @lcpr-template-start
#include <bits/stdc++.h>
using namespace std;
// @lcpr-template-end
// @lc code=start
#include <ranges>

class Solution {
public:
    int findMaxConsecutiveOnes(vector<int>& nums) {
        int n = nums.size();
        int ans = 0;
        for (int i = 0; i < n;) {
            while (i < n && nums[i] != 1) i++;
            int j = i;
            while (i < n && nums[i] == 1) i++;
            ans = max(ans, i - j);
        }
        return ans;
    }
};
// @lc code=end

File: Leetcode/rust/easy/485.rs
/*
 * @lc app=leetcode id=485 lang=rust
 *
 * [485] Max Consecutive Ones
 */


// @lcpr-template-start
struct Solution;
// @lcpr-template-end
// @lc code=start
impl Solution {
    pub fn find_max_consecutive_ones(nums: Vec<i32>) -> i32 {
        let n = nums.len();
        let mut ans = 0;
        let mut i = 0;
        while i < n {
            while i < n && nums[i] != 1 { i += 1; }
            let j = i;
            while i < n && nums[i] == 1 { i += 1; }
            ans = ans.max(i - j);
        }
        ans as i32
    }
}
// @lc code=end

